### ISSUE TITLE
PHP package 3.5 incompatible with pecl extension 3.4

### ISSUE THREAD

**Issue by michaelbausor** [google/protobuf#3881]
Installing PHP protobuf composer package 3.5 and pecl protobuf package 3.4, and then running this code:
```php
$a = new \Google\Protobuf\Any();
```

Produces this error:
```
Fatal error: Cannot use Google\Protobuf\Internal\DescriptorPool as DescriptorPool because the name is already in use in /usr/local/google/home/michaelbausor/dev/veneer/testing/protobuf-install/vendor/google/protobuf/php/src/Google/Protobuf/Any.php on line 7
```

cc @TeBoring

**Comment by TeBoring** [comment]
In 3.5，we provide all well known types in runtime. You need to remove the
files you generated in your library.

**Comment by michaelbausor** [comment]
This bug is unrelated to gax-php - it occurs in an empty project.

**Comment by michaelbausor** [comment]
This can be reproduced using these files:
- [composer.json](https://gist.github.com/michaelbausor/4eebe876f5b9bd66d15f550b0f65e76c)
- [test.php](https://gist.github.com/michaelbausor/995b8a8c5828c52228c8dfc7c11741b4)

**Comment by TeBoring** [comment]
Which line does the error happen?

**Comment by michaelbausor** [comment]
Output with xdebug enabled:

Fatal error: Cannot use Google\Protobuf\Internal\DescriptorPool as DescriptorPool because the name is already in use in /usr/local/google/home/michaelbausor/dev/veneer/testing/protobuf-install/vendor/google/protobuf/php/src/Google/Protobuf/Any.php on line 7

Call Stack:
    0.0002     259488   1. {main}() /usr/local/google/home/michaelbausor/dev/veneer/testing/protobuf-install/test.php:0
    0.0017     356200   2. spl_autoload_call() /usr/local/google/home/michaelbausor/dev/veneer/testing/protobuf-install/test.php:5
    0.0017     356288   3. Composer\Autoload\ClassLoader->loadClass() /usr/local/google/home/michaelbausor/dev/veneer/testing/protobuf-install/test.php:5
    0.0019     356960   4. Composer\Autoload\includeFile() /usr/local/google/home/michaelbausor/dev/veneer/testing/protobuf-install/vendor/composer/ClassLoader.php:301

**Comment by TeBoring** [comment]
BTW, does remove dependency on protonic in composer helps?

**Comment by michaelbausor** [comment]
Removing composer dependency on `google/protobuf` causes a different error, because the extension is 3.4 so does not provide well known types:

Fatal error: Class 'Google\Protobuf\Any' not found in /usr/local/google/home/michaelbausor/dev/veneer/testing/protobuf-install/test.php on line 5

Call Stack:
    0.0002     259488   1. {main}() /usr/local/google/home/michaelbausor/dev/veneer/testing/protobuf-install/test.php:0

**Comment by TeBoring** [comment]
3.5 will be released on pecl soon.

**Comment by bshaffer** [comment]
We have both `Google\Protobuf\Internal\DescriptorPool` and `Google\Protobuf\DescriptorPool` in the same repository, and these are conflicting since `Google\Protobuf\Any` we reference the internal class, but the other class exists in the same namespace.

**Comment by TeBoring** [comment]
Does use the fully qualified name inline help?

**Comment by TeBoring** [comment]
Please help try use fully qualified name in vendor/google/protobuf/php/src/Google/Protobuf/Any.php. If that works, I can make a patch.

**Comment by michaelbausor** [comment]
Removing `use Google\Protobuf\Internal\DescriptorPool;` and using fully qualified name does seem to fix the problem

**Comment by bshaffer** [comment]
@TeBoring yes, using the fully-quantified namespace for `DescriptorPool` will fix the issue.

I compared the namespaces and these are the classes which need to be fully qualified in order to fix the issue:

 - `Descriptor`
 - `DescriptorPool`
 - `EnumDescriptor`
 - `FieldDescriptor`

**Comment by TeBoring** [comment]
A question: isn't use statement only visible in the file of declaration? https://stackoverflow.com/questions/31770297/how-to-include-php-aliases-across-files

**Comment by TeBoring** [comment]
I think the reason is Any is in the namespace of Google/Protobuf. This causes the php first try to find ```Google\Protobuf\DescriptorPool``` when meeting DescritprorPool. So we only need to fix classes in Google\Protobuf namespace which also use Descriptor|DescriptorPool|... in Google\Protobuf\Internal namespace.

**Comment by TeBoring** [comment]
Fixed in #3886
